Cloud9 Support for Layered Stacks (#224)
aws-samples/cfn101-workshop: content/intermediate/templates/layered-stacks/index.md

```

##### 2. Deploy the VPC Stack
   :::::tabs{variant="container"}
	::::tab{id="cloud9" label="Cloud9"}
  1. In the **Cloud9 terminal** navigate to `cfn101-workshop/code/workspace/layered-stacks`.
  1. **Create Stack** by using the following AWS CLI command. The template requires you provide the values for `AvailabilityZones` parameter, For example `us-east-1a` and `us-east-1b` are used below. Please select 2 Availability Zone based on your region. 
  :::code{language=shell showLineNumbers=false showCopyAction=true}
  aws cloudformation create-stack \

>     --stack-name cfn-workshop-vpc \

Reviewer: ```suggestion
    --stack-name cfn-workshop-layered-stacks-vpc \
```

    --template-body file://vpc.yaml \
    --parameters ParameterKey=AvailabilityZones,ParameterValue=us-east-1a\\,us-east-1b
  :::
  1. Wait until the stack creation is completed by running the following AWS CLI command
  :::code{language=shell showLineNumbers=false showCopyAction=true}
  aws cloudformation wait stack-create-complete \

>     --stack-name cfn-workshop-vpc

Reviewer: ```suggestion
    --stack-name cfn-workshop-layered-stacks-vpc
```

  :::
  ::::
	::::tab{id="local" label="Local development"}
  1. Navigate to CloudFormation in the console and click **Create stack With new resources (standard)**.
  1. In **Prepare template** select **Template is ready**.
  1. In **Template source** select **Upload a template file**.
  1. Choose a file `vpc.yaml`.

>   1. Enter a **stack name**. For example, `cfn-workshop-vpc`.

Reviewer: ```suggestion
  1. Enter a **stack name**. For example, `cfn-workshop-layered-stacks-vpc`.
```

  1. For the **AvailabilityZones** parameter, select **2 AZs**.
  1. You can leave the rest of the parameters **default**.
  1. Navigate through the wizard leaving everything default.

>   1. On the Review page, scroll down to the bottom and click on **Create stack**.

Reviewer: ```suggestion
  1. On the Review page, scroll down to the bottom and click on **Submit**.
```

    ```

##### 2. Deploy the IAM Stack
   :::::tabs{variant="container"}
	::::tab{id="cloud9" label="Cloud9"}
  1. Let's **Create Stack** by using the following AWS CLI command. The template requires you to specify `CAPABILITY_IAM` capability for creating IAM resources.
  :::code{language=shell showLineNumbers=false showCopyAction=true}
  aws cloudformation create-stack \

>     --stack-name cfn-workshop-iam \

Reviewer: ```suggestion
    --stack-name cfn-workshop-layered-stacks-iam \
```

    --template-body file://iam.yaml \
    --capabilities CAPABILITY_IAM
  :::
  1. Wait until the stack creation is completed by running the following AWS CLI command
  :::code{language=shell showLineNumbers=false showCopyAction=true}
  aws cloudformation wait stack-create-complete \

>     --stack-name cfn-workshop-iam

Reviewer: ```suggestion
    --stack-name cfn-workshop-layered-stacks-iam
```

  :::
  ::::
	::::tab{id="local" label="Local development"}
  1. Navigate to CloudFormation in the console and click **Create stack With new resources (standard)**.
  1. In **Prepare template** select **Template is ready**.
  1. In **Template source** select **Upload a template file**.
  1. Choose a file `iam.yaml`.

>   1. Enter a **stack name**. For example, `cfn-workshop-iam`.

Reviewer: ```suggestion
  1. Enter a **stack name**. For example, `cfn-workshop-layered-stacks-iam`.
```

  1. Click **Next**.
  1. Navigate through the wizard leaving everything default.

>   1. **Acknowledge IAM capabilities** and click on **Create stack**.

Reviewer: ```suggestion
  1. **Acknowledge IAM capabilities** and click on **Submit**.
```

```

##### 5. Deploy the EC2 Stack
   :::::tabs{variant="container"}
	::::tab{id="cloud9" label="Cloud9"}
  1. Let's **Create Stack** by using the following AWS CLI command. The template requires you to specify `CAPABILITY_IAM` capability for creating IAM resources.
  :::code{language=shell showLineNumbers=false showCopyAction=true}
  aws cloudformation create-stack \

>     --stack-name cfn-workshop-ec2 \

Reviewer: ```suggestion
    --stack-name cfn-workshop-layered-stacks-ec2 \
```

    --template-body file://ec2.yaml 
  :::
  1. Wait until the stack creation is completed by running the following AWS CLI command
  :::code{language=shell showLineNumbers=false showCopyAction=true}
  aws cloudformation wait stack-create-complete \

>     --stack-name cfn-workshop-ec2

Reviewer: ```suggestion
    --stack-name cfn-workshop-layered-stacks-ec2
```

  :::
  ::::
	::::tab{id="local" label="Local development"}
  1. Navigate to CloudFormation in the console and click **Create stack With new resources (standard)**.
  1. In **Prepare template** select **Template is ready**.
  1. In **Template source** select **Upload a template file**.
  1. Choose a file `ec2.yaml`.

>   1. Enter a **stack name**. For example, `cfn-workshop-ec2`.

Reviewer: ```suggestion
  1. Enter a **stack name**. For example, `cfn-workshop-layered-stacks-ec2`.
```

  1. You can leave the rest of the parameters **default**.
  1. Navigate through the wizard leaving everything default.

>   1. On the **Review page**, scroll down to the bottom and click on **Create stack**.

Reviewer: ```suggestion
  1. On the **Review page**, scroll down to the bottom and click on **Submit**.
```

For example, you can not delete the **VPC stack** before you delete **EC2 stack**. You get following error message:

![delete-export-before-import.png](/static/intermediate/templates/layered-stacks/delete-export-before-import.png)
   :::::tabs{variant="container"}
	::::tab{id="cloud9" label="Cloud9"}
  1. **Delete Stack** by running the following AWS CLI command
  :::code{language=shell showLineNumbers=false showCopyAction=true}
  aws cloudformation delete-stack \

>     --stack-name cfn-workshop-ec2

Reviewer: ```suggestion
    --stack-name cfn-workshop-layered-stacks-ec2
```

  :::
  1. Wait until the stack is deleted by using the following AWS CLI command.
  :::code{language=shell showLineNumbers=false showCopyAction=true}
  aws cloudformation wait stack-delete-complete \
    --stack-name cfn-workshop-ec2
  :::

>   1. Repeat steps (1-2) above for stacks: `cfn-workshop-iam` and `cfn-workshop-vpc2`.

Reviewer: ```suggestion
  1. Repeat steps (1-2) above for stacks: `cfn-workshop-layered-stacks-iam` and `cfn-workshop-layered-stacks-vpc`.
```

  ::::
	::::tab{id="local" label="Local development"}

>   1. In the **[CloudFormation console](https://console.aws.amazon.com/cloudformation)**, select the **EC2 stack**, for example `cfn-workshop-ec2`.

Reviewer: ```suggestion
  1. In the **[CloudFormation console](https://console.aws.amazon.com/cloudformation)**, select the **EC2 stack**, for example `cfn-workshop-layered-stacks-ec2`.
```

  1. In the top right corner, click on **Delete**.

>   1. In the pop-up window click on **Delete stack**.

Reviewer: ```suggestion
  1. In the pop-up window click on **Delete**.
```
